transforms/tabular/tabular-merger-plugin/src/polus/plugins/transforms/tabular/tabular_merger/tabular_merger.py

```diff
@@ -0,0 +1,289 @@
+"""Tabular Merger."""
+import enum
+import functools as ft
+import logging
+import os
+import pathlib
+from collections import Counter
+from typing import List, Optional, Union
+
+import numpy as np
+import vaex
+from tqdm import tqdm
+
+logger = logging.getLogger(__name__)
+
+POLUS_TAB_EXT = os.environ.get("POLUS_TAB_EXT", ".csv")
+
+
+class Extension(str, enum.Enum):
+    """File format of an output combined file."""
+
+    CSV = ".csv"
+    ARROW = ".arrow"
+    PARQUET = ".parquet"
+    HDF = ".hdf5"
+    FEATHER = ".feather"
+    Default = "default"
+
+
+class Dimensions(str, enum.Enum):
+    """File format of an output combined file."""
+
+    Rows = "rows"
+    Columns = "columns"
+    Default = "default"
+
+
+def sorted_dataframe_list(
+    x: List[vaex.dataframe.DataFrameLocal],
+) -> List[vaex.dataframe.DataFrameLocal]:
+    """Reordering of list of dataframes based on the size.
+
+    Args:
+        x: List of vaex dataframes.
+    Returns:
+        sorted list of vaex dataFrame based on the size.
+    """
+    my_dict = {k: v for k, v in zip(x, [x[i].shape[0] for i in range(len(x))])}
+    occurrences = {k: v for k, v in Counter(my_dict.values()).items()}
+
+    for k, v in my_dict.items():
+        count = occurrences[v]
+        if count > 1:
+            increment = v + 1
+            my_dict[k] = increment
+            occurrences[v] = count - 1
+
+    sorted_values = sorted(my_dict.values(), reverse=True)
+    i = 0
+    status = "Unknown"
+    prf = []
+    while status != "true":
+        for o in sorted_values:
+            for k, v in my_dict.items():
+                if v == o:
+                    status = "true"
+                    prf.append(k)
+                    i += 1
+
+    return prf
+
+
+def remove_files(curr_dir: pathlib.Path) -> None:
+    """Delete intermediate hdf5 and yaml files in a working directory.
+
+    Args:
+        curr_dir: Path to the working directory.
+    """
+    for f in curr_dir.iterdir():
+        if f.suffix in [".hdf5", ".yaml"]:
+            os.remove(f)
+
+
+def merge_files(
+    inp_dir_files: List,
+    strip_extension: bool,
+    file_extension: Union[Extension, str],
+    dim: Union[Dimensions, str],
```

Comment:
The types should only be `Extension` and `Dimensions` respectively. Remove the union with `str`

Reply:
Done